pramen/core/src/main/scala/za/co/absa/pramen/core/sql/SqlGeneratorBase.scala

```diff
@@ -24,18 +28,33 @@ package za.co.absa.pramen.core.sql
 abstract class SqlGeneratorBase(sqlConfig: SqlConfig) extends SqlGenerator {
   import SqlGeneratorBase._
 
-  /** This Wraps the column name with escaping characters. The escape logic varies among SQL dialects. */
-  def wrapIdentifier(identifier: String): String
+  /**
+    * This returns characters used for escaping into a mode that allows special characters in identifiers.
+    * For example,
+    *  - in Hive, column name 'my column' should be escaped with back quotes `my column`, in MS SQL Server
+    *  - in MS SQL server square braces are used instead [my column].
+    */
+  def beginEndEscapeChars: (Char, Char)
 
-  /** This validates and escapes the column name is needed. Escaping does not happen always to maintain backwards compatibility. */
-  final override def escapeIdentifier(identifier: String): String = {
-    if (sqlConfig.escapeIdentifiers) {
-      validateIdentifier(identifier)
+  def quoteSingleIdentifier(identifier: String): String = {
+    val (escapeBegin, escapeEnd) = beginEndEscapeChars
 
-      if (identifierNeedsEscaping(identifier)) {
-        identifier.split('.').map(wrapIdentifier).mkString(".")
-      } else
-        identifier
+    if (identifier.startsWith(s"$escapeBegin") && identifier.endsWith(s"$escapeEnd")) {
+      identifier
+    } else {
+      s"$escapeBegin$identifier$escapeEnd"
+    }
+  }
+
+  final def quote(identifier: String): String = {
```

Comment:
Actually, I missed `override` here since this method is in the trait. So it is a mistake, just a different kind 😆 

I want to make unconditional quoting to be available in the interface so that if somebody needs it, they don't have to rely on config option.